Core/Barcelona/IMCoreHelpers/Query.swift

```diff
@@ -0,0 +1,254 @@
+//
+//  Query.swift
+//  grapple
+//
+//  Created by Eric Rabil on 11/24/21.
+//
+
+import Foundation
+import SwiftCLI
+import IMCore
+
+public struct EncodingBox: Encodable {
+    public var ref: (Encoder) throws -> ()
+    
+    public init(encodable: Encodable) {
+        ref = encodable.encode(to:)
+    }
+    
+    public func encode(to encoder: Encoder) throws {
+        try ref(encoder)
+    }
+}
+
+@_spi(EncodingBoxInternals) public var visited: Set<NSObject> = Set()
+
+// Look up a bunch of methods/impls on NSInvocation
+let nsInvocationClass: AnyClass = NSClassFromString("NSInvocation")!
+
+// Look up the "invocationWithMethodSignature:" method
+let nsInvocationInitializer = unsafeBitCast(
+    method_getImplementation(
+        class_getClassMethod(nsInvocationClass, NSSelectorFromString("invocationWithMethodSignature:"))!
+    ),
+    to: (@convention(c) (AnyClass?, Selector, Any?) -> Any).self
+)
+
+// Look up the "setSelector:" method
+let nsInvocationSetSelector = unsafeBitCast(
+    class_getMethodImplementation(nsInvocationClass, NSSelectorFromString("setSelector:")),
+    to:(@convention(c) (Any, Selector, Selector) -> Void).self
+)
+
+let nsInvocationSetTarget = unsafeBitCast(
+    class_getMethodImplementation(nsInvocationClass, NSSelectorFromString("setTarget:")),
+    to: (@convention(c) (NSObject, Selector, NSObject) -> Void).self
+)
+
+let nsInvocationRetainArguments = unsafeBitCast(
+    class_getMethodImplementation(nsInvocationClass, NSSelectorFromString("retainArguments")),
+    to: (@convention(c) (NSObject, Selector) -> Void).self
+)
+
+let nsInvocationInvoke = unsafeBitCast(
+    class_getMethodImplementation(nsInvocationClass, NSSelectorFromString("invoke")),
+    to: (@convention(c) (NSObject, Selector) -> Void).self
+)
+
+let nsInvocationGetReturnValue = unsafeBitCast(
+    class_getMethodImplementation(nsInvocationClass, NSSelectorFromString("getReturnValue:")),
+    to: (@convention(c) (NSObject, Selector, UnsafeMutableRawPointer) -> Void).self
+)
+
+let invokeWithTargetSelector = NSSelectorFromString("invokeWithTarget:")
+var result: UnsafeMutableRawPointer = .allocate(byteCount: 0, alignment: 0)
+
+#if os(iOS)
+public extension NSObject {
+    var className: String {
+        NSStringFromClass(Self.self)
+    }
+}
+#endif
+
+extension NSObject: Encodable {
```

Comment:
Pretty sure this caused a crash at some point, just raising it here as I can't see if you changed the code